### DISCUSSION TITLE
Import from cooked.wiki

### DISCUSSION THREAD

**Discussion by matthewkdies** [mealie-recipes/mealie#3121] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Importing recipes for generic, “non-major” websites is not easy or feasible.

### Please provide a concise description of the feature that would resolve your issue.

Enabling imports from cooked.wiki, where most generic recipe links can be reduced to just ingredients and steps, would enable importing of almost any recipe link.

For example, this link is the result of adding a generic recipe link to cooked.wiki by preceding the recipe’s url with `cooked.wiki`: https://cooked.wiki/new?url=https%3A%2F%2Fministryofcurry.com%2Fvegetarian-pad-thai%2F

### Please consider and list out some caveats or tradeoffs made in your design decision

While technically it may be possible to use cooked.wiki in general (I don’t know if they have an API) to do generic importing as a fallback, it’d probably be simpler to just enable importing from an already parsed recipe’s URL. Then, if desired, users could further automate this and potentially submit their own PRs later on.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by eduardofcgo** [comment]
I am the developer of Cooked.wiki.
There is no official API yet. If there are recipes that mealie can't import, you can import them first into Cooked. Because I use very standard metadata for the recipes, mealie should be able to import any Cooked.wiki recipe.

**Comment by andrew-manger** [comment]
@eduardofcgo I can confirm that mealie cannot import cooked.wiki recipes. 

The main reason I use mealie over cooked is because of the UI. I love the metric/imperial conversion and diagramming of cooked, but the UI, organization, and lack of self hosting are essential. If you'd like any feedback on the UI, I'd be happy to donate my services to help improve it!

**Reply by eduardofcgo** [reply]
Hello,
Cooked.wiki recipes have the necessary metadata for most apps to be able to import them. I have not tested on Mealie specifically. It's likelly that the Cloudflare bot protection might be blocking Mealie.
Cooked.wiki does automatic categorization and generation of metadata, but also allows custom groups of recipes. I agree that the UI needs some improvements. Please feel free to get in touch, a fresh prespective can be very usefull. There is an app under development, which is open source.
